packages/dapp-toolkit/src/wallet-request/transport/radix-connect-relay/rcr-api-client.ts

```diff
@@ -0,0 +1,213 @@
+  const logger = input.logger?.getSubLogger({ name: 'RadixConnectRelayApi' })
+  const encryptionClient = input.providers.encryptionClient
+
+  const decryptResponse = (
+    secretHex: string,
+    value: string,
+  ): ResultAsync<WalletInteractionResponse, SdkError> =>
+    transformBufferToSealbox(Buffer.from(value, 'hex'))
+      .asyncAndThen(({ ciphertextAndAuthTag, iv }) =>
+        encryptionClient.decrypt(
+          ciphertextAndAuthTag,
+          Buffer.from(secretHex, 'hex'),
+          iv,
+        ),
+      )
+      .andThen((decrypted) =>
+        parseJSON<WalletInteractionResponse>(decrypted.toString('utf-8')),
+      )
+      .mapErr(() => SdkError('FailedToDecrypt', ''))
+
+  const encryptWalletInteraction = (
+    walletInteraction: WalletInteraction,
+    secret: Buffer,
+  ): ResultAsync<string, SdkError> =>
+    encryptionClient
+      .encrypt(Buffer.from(JSON.stringify(walletInteraction), 'utf-8'), secret)
+      .mapErr(() =>
+        SdkError(
+          'FailEncryptWalletInteraction',
+          walletInteraction.interactionId,
+        ),
+      )
+      .map((sealedBoxProps) => sealedBoxProps.combined.toString('hex'))
+
+  const sendRequest = (
+    { sessionId, sharedSecret }: ActiveSession,
+    walletInteraction: WalletInteraction,
+  ): ResultAsync<void, SdkError> => {
+    logger?.debug({ method: 'sendRequest', sessionId, walletInteraction })
+    return encryptWalletInteraction(
+      walletInteraction,
+      Buffer.from(sharedSecret, 'hex'),
+    ).andThen((encryptedWalletInteraction) =>
+      fetchWrapper(
+        fetch(baseUrl, {
+          method: 'POST',
+          body: JSON.stringify({
+            method: 'sendRequest',
+            sessionId,
+            data: encryptedWalletInteraction,
+          }),
+        }),
+      )
+        .map(() => undefined)
+        .mapErr(() => SdkError('FailedToSendRequestToRadixConnectRelay', '')),
+    )
+  }
+
+  const getResponses = (session: ActiveSession) =>
+    fetchWrapper<string[]>(
+      fetch(baseUrl, {
+        method: 'POST',
+        body: JSON.stringify({
+          method: 'getResponses',
+          sessionId: session.sessionId,
+        }),
+      }),
+    )
+      .mapErr(() => SdkError('FailedToGetRequestsFromRadixConnectRelay', ''))
+      .andThen((value) =>
+        ResultAsync.combine(
+          value.data.map((encryptedWalletInteraction) =>
+            decryptResponse(session.sharedSecret, encryptedWalletInteraction),
+          ),
+        ),
+      )
+
+  const sendHandshakeRequest = (
+    sessionId: string,
+    publicKeyHex: string,
+  ): ResultAsync<void, SdkError> => {
+    logger?.debug({ method: 'sendHandshakeRequest', sessionId, publicKeyHex })
+    return fetchWrapper(
+      fetch(baseUrl, {
+        method: 'POST',
+        body: JSON.stringify({
+          method: 'sendHandshakeRequest',
+          sessionId,
+          data: publicKeyHex,
+        }),
+      }),
+    )
+      .map(() => {
+        logger?.debug({
```

Comment:
Some methods have `logger` statements others don't. Maybe let's unify it by puting `logger` calls into another fetch wrapper method I mentioned in other comment or remove it as a whole

Reply:
refactored